File: test/org/rust/lang/core/parser/RustParsingTestCaseBase.java
package org.rust.lang.core.parser;

import com.intellij.testFramework.ParsingTestCase;
import org.jetbrains.annotations.NonNls;
import org.rust.lang.core.RustParserDefinition;

public abstract class RustParsingTestCaseBase extends ParsingTestCase {

    @Override
    protected String getTestDataPath() {
        return "test/testData";
    }

    @SuppressWarnings({"JUnitTestCaseWithNonTrivialConstructors"})
    public RustParsingTestCaseBase(@NonNls final String dataPath) {
        super("psi/" + dataPath, "rust", true /*lowerCaseFirstLetter*/, new RustParserDefinition());
    }

    @Override
    protected void setUp() throws Exception {
        super.setUp();
    }

    @Override
    protected void tearDown() throws Exception {
        super.tearDown();
    }

}

File: test/org/rust/lang/core/parser/RustParsingTestCaseBase.kt
package org.rust.lang.core.parser

import com.intellij.lang.LanguageBraceMatching
import com.intellij.testFramework.ParsingTestCase
import org.jetbrains.annotations.NonNls
import org.rust.lang.RustLanguage
import org.rust.lang.core.RustParserDefinition
import org.rust.lang.highlight.RustBraceMatcher

abstract class RustParsingTestCaseBase(@NonNls dataPath: String)
 : ParsingTestCase("psi/" + dataPath, "rs", true /*lowerCaseFirstLetter*/, RustParserDefinition()) {

    override fun getTestDataPath(): String = "testData"

    override fun setUp() {
        super.setUp()
        addExplicitExtension(LanguageBraceMatching.INSTANCE, RustLanguage.INSTANCE, RustBraceMatcher())
    }

    override fun tearDown() {
        super.tearDown()
    }
}
